apps/gql/loaders/pano.ts

```diff
@@ -200,9 +202,21 @@ export const transformPanoCommentConnection = (connection: Connection<Comment>)
   } satisfies PanoCommentConnection;
 };
 
-export const transformPanoUpvote = (upvote: Upvote) => {
+export function isPostUpvote(upvote: Upvote | CommentUpvote): upvote is Upvote {
+  return !!(upvote as Upvote).postID;
+}
+
+export function isCommentUpvote(upvote: Upvote | CommentUpvote): upvote is Upvote {
+  return !!(upvote as CommentUpvote).commentID;
+}
+
+export const transformPanoUpvote = (upvote: Upvote | CommentUpvote) => {
+  const type = isPostUpvote(upvote) ? "PanoPost" : "PanoComment";
+  const id = stringify(type, upvote.id);
```

Comment:
we do this over here to figure out upvotable type from global id. It's messy, but it gets the job done